### PR TITLE
Add Node PM plugin

### PR DESCRIPTION
**Pull request by qwerty287**
A simple plugin that executes NPM, PNPM or Yarn scripts. We can also switch our pipeline to it.

### CODE REVIEW

File: docs/plugins/woodpecker-plugins/plugins.json [hunk 1]
```diff
@@ -45,6 +45,11 @@
       "docs": "https://raw.githubusercontent.com/appleboy/drone-git-push/master/DOCS.md",
       "verified": false
     },
+    {
+      "name": "Node PM",
+      "docs": "https://codeberg.org/qwerty287/woodpecker-node-pm/src/branch/main/docs.md",
```
Comment (6543) [inline docs/plugins/woodpecker-plugins/plugins.json:50]:
```suggestion
      "docs": "https://codeberg.org/qwerty287/woodpecker-node-pm/raw/branch/main/docs.md",
```

**Comment by lafriks** [comment]
just an idea but maybe it would be worth to move this plugin to https://codeberg.org/woodpecker-plugins/node-pm ?

**Comment by lafriks** [comment]
@6543 what do you think?

**Comment by 6543** [comment]
ack

**Comment by 6543** [comment]
@qwerty287 move accepted .. I'm adjusting settings now

**Comment by 6543** [comment]
https://codeberg.org/woodpecker-plugins/node-pm/pulls/1

**Comment by 6543** [comment]
#29 exporting to image
#29 21.80 error: failed commit on ref "layer-sha256:a9c1c41dae57fc1bdca998f4927abd6d90407ccd5dcd326522dee7411cc8b864": unexpected status: 500 Internal Server Error
#29 21.80 retrying in 1s
#29 21.82 error: failed commit on ref "layer-sha256:213ec9aee27d8be045c6a92b7eac22c9a64b44558193775a1a7f626352392b49": unexpected status: 500 Internal Server Error


we get hit by some gitea bug recently a lot :/

**Comment by 6543** [comment]
https://ci.codeberg.org/woodpecker-plugins/node-pm/pipeline/18/3

**Comment by 6543** [comment]
proposal: we can dual publish it for now so we can work with it ...

@qwerty287 wana file a pull?

I'll add the secrets same as with the buildx plugin

**Comment by 6543** [comment]
https://hub.docker.com/repository/docker/woodpeckerci/plugin-node-pm

**Review by 6543 — APPROVED** [review]

**Pull request merged**